lib/couchbase/utils/time.rb

```diff
@@ -48,8 +48,20 @@ def extract_expiry_time(time_or_duration)
         end
       end
 
+      # This method converts its argument to milliseconds
+      #
+      # 1. Integer values are interpreted as a number of milliseconds
+      # 2. If the argument is a Duration-like object and responds to #in_milliseconds,
+      #    then use it and convert result to Integer
+      # 3. Otherwise invoke #to_i on the argument and interpret it as a number of milliseconds
       def extract_duration(number_or_duration)
-        number_or_duration.respond_to?(:in_milliseconds) ? number_or_duration.public_send(:in_milliseconds) : number_or_duration
+        return number_or_duration if number_or_duration.is_a? Integer
+
+        if number_or_duration.respond_to?(:in_milliseconds)
+          number_or_duration.public_send(:in_milliseconds)
+        else
+          number_or_duration
+        end.to_i
```

Comment:
This breaks all the things, unfortunately.  I've monkeypatched this method in our local setup and the result of converting everything using `.to_i` means that `nil` values never make it out.  This method will always return an `Integer`.  So if the client specifies _nothing at all_ for the timeout the default value for the timeout will not be used.

Locally, it means that every attempt to connect to the cluster results in this:

```       
Failure/Error: cluster = Cluster.connect("couchbase://#{host}?enable_dns_srv=false", couch_options)
       
Couchbase::Error::UnambiguousTimeout:
unable open cluster at 127.0.0.1: unambiguous_timeout (14)
```

I've changed this locally as follows:

```ruby
        result = if number_or_duration.respond_to?(:in_milliseconds)
          number_or_duration.public_send(:in_milliseconds)
        else
          number_or_duration
        end.to_i

        result.nil? ? result : result.to_i
```

Reply:
Tagging @avsej and @DemetrisChr just to make sure this comment wasn't missed.  Seems like a potentially serious issue.

Reply:
hi @mikesnare, sorry I didn't notice comment in closed PR. I've fixed nil condition in this PR: https://github.com/couchbase/couchbase-ruby-client/pull/111